server/app/controllers/fusor/api/customer_portal/customer_portal_proxies_controller.rb

```diff
@@ -32,6 +32,15 @@ def logout
           render :json => {}
         end
 
+        def is_authenticated
+          authenticated = false
+          if session[:portal_username] && session[:portal_password]
+            # they are not nil, let's make sure neither is empty
+            authenticated = session[:portal_username].empty? && session[:portal_password].empty?
```

Comment:
For that to happen, both session[:portal_username] and session[:portal_password] would have to be empty string and not nil.  Pretty unlikely.  Could write a unit test for those scenarios, if you feel it's worth it.